sale_order_project/tests/test_sale_order_project.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2016 Onestein (<http://www.onestein.eu>)
+# Copyright 2018 Aitor Bouzas (<http://www.adaptivecity.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import odoo.tests.common as common
+from odoo.exceptions import UserError
+
+
+class TestSaleOrderProject(common.TransactionCase):
+
+    def setUp(self):
+        super(TestSaleOrderProject, self).setUp()
+        self.sale_order_model = self.env['sale.order']
+        self.analytic_account_model = self.env['account.analytic.account']
+        self.analytic_account = self.env.ref('account.analytic_account_3')
+        self.sale_order = self.env.ref('sale.sale_order_2')
+        self.project = self.env.ref('project.project_project_1')
+
+    def test_create_project_from_order(self):
+        self.sale_order.action_create_project()
+        project_dict = self.sale_order_model._prepare_project_vals(
+            self.sale_order)
+        project_name = self.sale_order.related_project_id.name
+        self.assertEqual(self.sale_order.partner_id.id,
+                         project_dict['partner_id'])
+        self.assertEqual(project_name, project_dict['name'])
+        self.assertEqual(self.sale_order.user_id.id, project_dict['user_id'])
+
+    def test_assigned_analytic_account_project(self):
+        self.sale_order.write({
+            'analytic_account_id': self.project.analytic_account_id.id
+        })
+        self.assertEqual(self.sale_order.related_project_id,
+                         self.project)
+
+    def test_compute_sale_orders_count(self):
+        self.sale_order.analytic_account_id = \
+            self.project.analytic_account_id.id
+        self.assertTrue(self.project.sale_order_count, 1)
+
+    def test_sale_order_tree_view(self):
+        res = self.project.sale_order_tree_view()
+        self.assertTrue(res)
+
+    def test_error_creating_project(self):
+        with self.assertRaises(UserError):
+            self.sale_order.action_create_project()
+            self.sale_order.action_create_project()
+
+    def test_compute_num_sale_orders(self):
+        self.sale_order.analytic_account_id = \
+            self.project.analytic_account_id.id
+        self.assertTrue(self.analytic_account.num_sale_orders, 1)
+
+    def test_action_create_project_task_0(self):
+        self.wizard.action_create_project()
+
+    def test_error_action_create_project_task(self):
+        with self.assertRaises(UserError)
```

Comment:
https://travis-ci.org/OCA/project/jobs/441667575#L1725
```bash
  File "/home/travis/build/OCA/project/sale_order_project/tests/test_sale_order_project.py", line 59
    with self.assertRaises(UserError)
                                    ^
SyntaxError: invalid syntax
```

You need to add `:` at final of this sentence.
and a mute_logger will be needed too, here.